src/fuzzfetch/fetch.py

```diff
@@ -349,9 +358,51 @@ def __init__(self, target, branch, build, flags, platform=None):
             self._task = build
 
         else:
-            self._task = BuildTask(build, branch, self._flags, self._platform)
-
             now = datetime.now(timezone('UTC'))
+            try:
+                self._task = BuildTask(build, branch, self._flags, self._platform)
+            except FetcherException:
+                if not nearest:
+                    raise
+                else:
+                    asc = nearest == Fetcher.BUILD_ORDER_ASC
+                    start = None
+                    if build == 'latest':
+                        start = now
+                    elif re.match(r'^[0-9[a-f]{12,40}$', build):
+                        start = _get_rev_date(build)
+
+                    elif re.match(r'^[0-9]{4}-[0-9]{2}-[0-9]{2}$', build):
+                        date = datetime.strptime(build, '%Y-%m-%d')
+                        localized = timezone('UTC').localize(date)
+                        start = localized + timedelta(days=1) if asc else localized - timedelta(days=1)
```

Comment:
Why is this +/- 1? You're not doing that in the namespace case below.

Reply:
In both rev and namespace cases, we're looking for a specific match.  There's still a chance that a build exists for that push date so we don't increment the current day.  Whereas with pushdate, we've already tried and failed on the current date.